lib/spack/spack/compiler.py

```diff
@@ -294,6 +294,8 @@ def determine_implicit_rpaths(cls, paths):
                                       output=str, error=str))  # str for py2
 
             return cls.parse_implicit_rpaths(output)
+        except spack.util.executable.ProcessError:
+            tty.debug('ProcessError: Command exited with non-zero status')
```

Comment:
@adamjstewart two requests:

* can you include the error message in the output (i.e. retrieve it from `ProcessError.long_message`) 
* can you return a `[]` to keep the return type consistent (i.e. so callers can always iterate over the result)?